internal/selfupdate/selfupdate.go

```diff
@@ -0,0 +1,341 @@
+package selfupdate
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	wailsruntime "github.com/wailsapp/wails/v2/pkg/runtime"
+)
+
+// Version is the current version of the application. Set at compile time for production builds using ldflags (see tasks in the /tasks/build directory).
+var Version = "development"
+
+// noSelfUpdate is set to "true" for builds distributed to package managers to prevent auto-updating. It is typed as a string because the linker allows only setting string variables at compile time (see https://pkg.go.dev/cmd/link).
+// Set at compile time using ldflags (see the prod-noupdate task in the /tasks/build directory).
+var noSelfUpdate = "false"
+
+// releaseTrack is the release track to follow for updates. It currently only takes the value "stable".
+var releaseTrack = "stable"
+
+// manifestsBaseURL is the base URL for fetching update manifests.
+const manifestsBaseURL = "https://zenprivacy.net/update-manifests"
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type SelfUpdater struct {
+	version      string
+	noSelfUpdate bool
+	releaseTrack string
+	httpClient   HTTPClient
+}
+
+type Release struct {
```

Comment:
Probably should be a private struct.

Reply:
Alright, I had thought that all structs needed to be capitalized for some reason.